src/core/util/utils.cpp

```diff
@@ -674,20 +674,20 @@ class socket_context_manager {
             .tv_usec = 10,
         };
 
-        m_fd = orig_os_api.socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+        m_fd = SYSCALL(socket, AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
         if (m_fd < 0) {
             throw std::runtime_error("Open netlink socket failed");
         }
 
-        if (orig_os_api.setsockopt(m_fd, SOL_SOCKET, SO_RCVTIMEO, (const char *)&tv, sizeof tv)) {
-            close(m_fd);
+        if (SYSCALL(setsockopt, m_fd, SOL_SOCKET, SO_RCVTIMEO, (const char *)&tv, sizeof tv)) {
+            SYSCALL(close, m_fd);
```

Comment:
Shouldn't be XLIO_CALL?

Reply:
since we opened the socket with `orig_os_api.socket` / `SYSCALL` the expected behavior is to use a similar close.